raft/raft.go

```diff
@@ -1381,7 +1387,15 @@ func (r *raft) handleAppendEntries(m pb.Message) {
 	} else {
 		r.logger.Debugf("%x [logterm: %d, index: %d] rejected MsgApp [logterm: %d, index: %d] from %x",
 			r.id, r.raftLog.zeroTermOnErrCompacted(r.raftLog.term(m.Index)), m.Index, m.LogTerm, m.Index, m.From)
-		r.send(pb.Message{To: m.From, Type: pb.MsgAppResp, Index: m.Index, Reject: true, RejectHint: r.raftLog.lastIndex()})
+		reject := r.raftLog.lastIndex()
+		if m.Index < reject {
+			reject = m.Index
+		}
+		logTerm, _ := r.raftLog.term(reject)
+		r.send(pb.Message{To: m.From, Type: pb.MsgAppResp, Index: m.Index, Reject: true,
+			RejectHint: reject,
+			LogTerm:    logTerm,
```

Comment:
Please add a comment on `LogTerm` (in the proto) that in a MsgAppResp, it is only set on rejections, and to the term corresponding to RejectHint. This is necessary because the naive assumption would be that it corresponds to `Index`.